examples/AffinePlaneCurve.jl

```diff
@@ -384,14 +396,229 @@ function curve_singular_locus(C::AffinePlaneCurve)
            return [CC, Pts]
         else
            M = curve_intersect(C, S[1][1])
-           CC = M[1]
+           MM = M[1]
            PP = [Pts; M[2]]
+           rCC = reduction(MM[1])
+           push!(CC, AffinePlaneCurve(rCC.eq*MM[1].eq))
            return [CC, PP]
         end
      end
   end
 end
 
+################################################################################
+# Compute squarefree equation.
+# TODO: change for a direct squarefree computation when available.
+
+@doc Markdown.doc"""
+reduction(C::AffinePlaneCurve)
+> Returns the affine plane curve defined by the squarefree part of the equation of C.
+"""
+function reduction(C::AffinePlaneCurve)
+  if C.components == Dict()
+     f = factor(C.eq)
+     C.components = Dict(AffinePlaneCurve(x) => f.fac[x] for x in keys(f.fac))
+  end
+  F = 1
+  for D in keys(C.components)
+     F = F*D.eq
+  end
+  rC = AffinePlaneCurve(F)
+  rC.components = Dict(AffinePlaneCurve(D.eq) => 1 for D in keys(C.components))
+  return(rC)
```

Comment:
Just a minor side remark: `return` is not a function; the idiomatic way to write this (also elsewhere) would be this:
```suggestion
  return rC
```